src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -6566,6 +6590,15 @@ private void ReportNullabilityMismatchWithTargetDelegate(Location location, Name
             }
 
             Debug.Assert(delegateType is object);
+
+            if (unboundLambda.HasExplicitReturnType(out _, out var returnType) &&
```

Comment:
Should we take the refKind into account too?

Reply:
We're already requiring an identity conversion so I don't think ref kind will have any effect. Added a test in case the identity conversion requirement is relaxed.